src/Components/WebAssembly/Server/src/ComponentsWebAssemblyApplicationBuilderExtensions.cs

```diff
@@ -82,5 +113,81 @@ private static void AddMapping(FileExtensionContentTypeProvider provider, string
                 provider.Mappings.Add(name, mimeType);
             }
         }
+
+        private static void NegotiateEncoding(HttpContext context, IWebHostEnvironment webHost)
+        {
+            var accept = context.Request.Headers[HeaderNames.AcceptEncoding];
+            if (StringValues.IsNullOrEmpty(accept))
+            {
+                return;
+            }
+
+            if (!StringWithQualityHeaderValue.TryParseList(accept, out var encodings) || encodings.Count == 0)
+            {
+                return;
+            }
+
+            var selectedEncoding = StringSegment.Empty;
+            var selectedEncodingQuality = .0;
+
+            foreach (var encoding in encodings)
+            {
+                var encodingName = encoding.Value;
+                var quality = encoding.Quality.GetValueOrDefault(1);
+
+                if (quality < double.Epsilon)
+                {
+                    continue;
+                }
+
+                if (quality <= selectedEncodingQuality)
+                {
+                    continue;
+                }
+
+                if (_supportedEncodings.Contains(encodingName))
+                {
+                    selectedEncoding = encodingName;
+                    selectedEncodingQuality = quality;
+                }
+
+                if (StringSegment.Equals("*", encodingName, StringComparison.Ordinal))
+                {
+                    foreach (var candidate in _preferredEncodings)
+                    {
+                        if (ResourceExists(context, webHost, candidate.extension))
+                        {
+                            selectedEncoding = candidate.encoding;
+                            break;
+                        }
+                    }
+
+                    selectedEncodingQuality = quality;
+                }
+
+                if (StringSegment.Equals("identity", encodingName, StringComparison.OrdinalIgnoreCase))
+                {
+                    selectedEncoding = StringSegment.Empty;
+                    selectedEncodingQuality = quality;
+                }
+            }
+
+            if (StringSegment.Equals("gzip", selectedEncoding, StringComparison.OrdinalIgnoreCase))
+            {
+                var targetPath = context.Request.Path + ".gz";
+                if (webHost.WebRootFileProvider.GetFileInfo(targetPath).Exists)
```

Comment:
Lines 177 and 178 could be replaced with:

```
if (ResourceExists(context, webHost, ".gz"))
```

Not an essential change, but hey you've got the utility method already :)

Reply:
Ah, I was avoiding recomputing the path twice :)

Reply:
Updated